packages/sitecore-jss-nextjs/src/components/NextImage.test.tsx

```diff
@@ -0,0 +1,323 @@
+          disableStaticImages: false,
+          minimumCacheTTL: 1,
+          formats: [],
+          dangerouslyAllowSVG: false,
+          contentSecurityPolicy: 'test',
+        },
+        src: 'https://cm.jss.localhost/assets/img/test0.png?mw=100',
+        width: 100,
+      };
+      const result = loader(params);
+      expect(result).to.be.a('string');
+      expect(result).to.equal('https://cm.jss.localhost/assets/img/test0.png?mw=100');
+    });
+
+    it('should not append config path when src is absolute', () => {
+      const params: ImageLoaderProps = {
+        config: {
+          deviceSizes: [],
+          imageSizes: [],
+          path: 'https://cm.jss.localhost/',
+          allSizes: [],
+          loader: 'default',
+          domains: [],
+          disableStaticImages: false,
+          minimumCacheTTL: 1,
+          formats: [],
+          dangerouslyAllowSVG: false,
+          contentSecurityPolicy: 'test',
+        },
+        src: 'https://cm.jss.localhost/assets/img/test0.png?mw=100',
+        width: 100,
+      };
+      const result = loader(params);
+      expect(result).to.be.a('string');
+      expect(result).to.equal('https://cm.jss.localhost/assets/img/test0.png?mw=100');
+    });
+
+    it('should throw an error if path is not configured', () => {
+      const params: ImageLoaderProps = {
+        config: {
+          deviceSizes: [],
+          imageSizes: [],
+          path: undefined,
+          allSizes: [],
+          loader: 'default',
+          domains: [],
+          disableStaticImages: false,
+          minimumCacheTTL: 1,
+          formats: [],
+          dangerouslyAllowSVG: false,
+          contentSecurityPolicy: 'test',
+        },
+        src: '/assets/img/test0.png?mw=100',
+        width: 100,
+      };
+      expect(() => loader(params)).to.throw(
+        'Failed to load image. Please make sure images path is configured correctly in next.config.js'
+      );
+    });
+  });
+
+  describe('error cases', () => {
+    const src = '/assets/img/test0.png';
+    it('should throw an error if src is present', () => {
+      expect(() => mount(<NextImage src={src} />)).to.throw(
+        'Detected src prop. If you wish to use src, use next/image directly.'
+      );
+    });
+  });
+
+  describe('with direct image object, no value/editable', () => {
+    const props = {
+      field: {
+        src: '/assets/img/test0.png',
+      },
+      width: 8,
+      height: 10,
+    };
+
+    const rendered = mount(<NextImage loader={mockLoader} {...props} />).find('Image');
+    it('should render image with url', () => {
+      rendered.render();
+      expect(rendered).to.have.lengthOf(1);
+      expect(rendered.prop('src')).to.equal(props.field.src);
+      expect(rendered.prop('width')).to.equal(props.width);
+      expect(rendered.prop('height')).to.equal(props.height);
+
+      expect(mockLoader.called).to.be.true;
+      expect(mockLoader).to.have.been.calledWith(loaderParams);
```

Comment:
Having the entire `loaderParams` seems unnecessary (and brittle - these defaults may change with Next.js version updates). Ideally we can get rid of `loaderParams` and assert only the properties we're interested in (i.e. src, width). We should be able to do this [using sinon.match](https://stackoverflow.com/a/30875810/233024).